Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ts

```diff
@@ -49,7 +49,9 @@ async function run() {
             tl.setVariable(webAppUri, publishingProfile.destinationAppUrl);
         }
 
-        if(useWebDeploy) {
+        var win = tl.osType().match(/^Win/);
+
+        if(useWebDeploy || win) {
```

Comment:
Can you create a function that can return true/false for msdeploy. At present you have two conditions and we are likely to add one more condition during support for Java apps. Will be clean if the decision making logic is in separate function.